modules/tinymce/src/plugins/table/test/ts/browser/TabKeyNavigationTest.ts

```diff
@@ -16,30 +18,51 @@ describe('browser.tinymce.plugins.table.TabKeyNavigationTest', () => {
     },
     base_url: '/project/tinymce/js/tinymce'
   }, [ Plugin, Theme ], true);
+  let events: Array<EditorEvent<TableModifiedEvent>> = [];
+  const logEvent = (event: EditorEvent<TableModifiedEvent>) => {
+    events.push(event);
+  };
+  afterEach(() => events = []);
 
   it('TBA: Tab key navigation', () => {
     const editor = hook.editor();
 
     editor.setContent('<table><tbody><tr><td>A1</td><td>A2</td></tr><tr><td>B1</td><td>B2</td></tr></tbody></table><p>x</p>');
 
     LegacyUnit.setSelection(editor, 'td', 0);
-    editor.fire('keydown', { keyCode: Keys.tab() } as KeyboardEvent);
+    TinyContentActions.keystroke(editor, Keys.tab());
     assert.equal(editor.selection.getStart().innerHTML, 'A2');
 
     LegacyUnit.setSelection(editor, 'td', 0);
-    editor.fire('keydown', { keyCode: Keys.tab(), shiftKey: true } as KeyboardEvent);
+    TinyContentActions.keystroke(editor, Keys.tab(), { shiftKey: true });
     assert.equal(editor.selection.getStart().innerHTML, 'A1');
 
     LegacyUnit.setSelection(editor, 'td:nth-child(2)', 0);
-    editor.fire('keydown', { keyCode: Keys.tab(), shiftKey: true } as KeyboardEvent);
+    TinyContentActions.keystroke(editor, Keys.tab(), { shiftKey: true });
     assert.equal(editor.selection.getStart().innerHTML, 'A1');
 
     LegacyUnit.setSelection(editor, 'tr:nth-child(2) td:nth-child(2)', 0);
-    editor.fire('keydown', { keyCode: Keys.tab() } as KeyboardEvent);
+    TinyContentActions.keystroke(editor, Keys.tab());
     assert.equal(editor.selection.getStart(true).nodeName, 'TD');
     TinyAssertions.assertContent(editor,
       '<table><tbody><tr><td>A1</td><td>A2</td></tr><tr><td>B1</td><td>B2' +
       '</td></tr><tr><td>&nbsp;</td><td>&nbsp;</td></tr></tbody></table><p>x</p>'
     );
   });
+
+  it('TINY-7006: Fire TableModified event when rows are added via the Tab key', () => {
+    const editor = hook.editor();
+
+    editor.on('TableModified', logEvent);
+
+    editor.setContent('<table><tbody><tr><td>A1</td><td>A2</td></tr><tr><td>B1</td><td>B2</td></tr></tbody></table><p>x</p>');
+
+    LegacyUnit.setSelection(editor, 'tr:nth-child(2) td:nth-child(2)', 0);
+    assert.lengthOf(events, 0);
+    TinyContentActions.keystroke(editor, Keys.tab());
+    TinyAssertions.assertContentPresence(editor, { tr: 3 });
+    assert.lengthOf(events, 1);
+    assert.equal(events[0].type, 'tablemodified');
+    editor.off('tablemodified', logEvent);
```

Comment:
Just for consistency 

```suggestion
    editor.off('TableModified', logEvent);
```